packages/deploy-config-sub-generator/src/app/index.ts

```diff
@@ -0,0 +1,467 @@
+    private readonly launchStandaloneFromYui: boolean;
+    private readonly abapChoice: Target = { name: TargetName.ABAP, description: 'ABAP' };
+    private readonly cfChoice: Target = { name: TargetName.CF, description: 'Cloud Foundry' };
+    private readonly apiHubConfig: ApiHubConfig;
+    private isLibrary = false;
+    private target?: string;
+    private indexGenerationAllowed = false;
+    private cfDestination: string;
+
+    /**
+     * Constructor for the deployment config generator.
+     *
+     * @param args - the arguments passed in
+     * @param opts - the options passed in
+     */
+    constructor(args: string | string[], opts: DeployConfigOptions) {
+        super(args, opts);
+        // this.env.adapter.promptModule is undefined when running in YUI
+        this.env.adapter.promptModule?.registerPrompt('autocomplete', autocomplete);
+        this.launchDeployConfigAsSubGenerator = opts.launchDeployConfigAsSubGenerator ?? false;
+        this.launchStandaloneFromYui = false; // By default
+        this.target = parseTarget(args, opts);
+
+        if (this.options.projectPath && this.options.projectName) {
+            this.options.appRootPath = join(this.options.projectPath, this.options.projectName);
+        } else {
+            this.options.appRootPath = this.destinationRoot();
+        }
+        this.options.projectRoot = this.options.appRootPath;
+
+        // Load .env file for api hub config
+        dotenv.config();
+
+        // Application Modeler launches Deployment Configuration Generator YUI.
+        // Pass project folder from command palette input during launching.
+        if (this.options.data?.destinationRoot) {
+            this.destinationRoot(this.options.data.destinationRoot);
+            this.launchDeployConfigAsSubGenerator = this.options.data.launchDeployConfigAsSubGenerator;
+            this.launchStandaloneFromYui = true;
+            this.options.appRootPath = join(dirname(this.destinationRoot()), basename(this.destinationRoot()));
+            this.options.projectRoot = this.destinationRoot();
+            dotenv.config({ path: join(this.destinationRoot(), '.env') });
+        }
+        this.apiHubConfig = this.options.apiHubConfig ?? getEnvApiHubConfig();
+    }
+
+    public async initializing(): Promise<void> {
+        await super.initializing();
+    }
+
+    public async prompting(): Promise<void> {
+        const { target, answers } = await this._prompting();
+
+        if ((answers as Answers)?.confirmConfigUpdate !== false && target) {
+            this._composeWithSubGenerator(target, answers);
+        } else {
+            DeploymentGenerator.logger?.debug(t('debug.exist'));
+            process.exit(0); // only relevant for CLI
+        }
+    }
+
+    /**
+     * Determines the target deployment and runs all prompting if required.
+     *
+     * @returns - target deployment and answers
+     */
+    private async _prompting(): Promise<{
```

Comment:
Please extract to another file? This will improve decouplng, testability and refactorability.